lib/java/com/google/android/material/slider/BaseSlider.java

```diff
@@ -489,7 +490,8 @@ private void validateValues() {
                 Float.toString(valueFrom),
                 Float.toString(valueTo)));
       }
-      if (stepSize > 0.0f && ((valueFrom - value) / stepSize) % 1 > THRESHOLD) {
```

Comment:
Doing `valueFrom - value` will always produce 0 or a negative value since `value >= valueFrom` which means that this check didn't actually validate the value and would allow any value.

I believe the correct check is `value - valueFrom`